library/core/src/array/mod.rs

```diff
@@ -504,7 +581,15 @@ where
     let mut guard: Guard<_, N> =
         Guard { ptr: MaybeUninit::slice_as_mut_ptr(&mut array), initialized: 0 };
 
-    while let Some(item) = iter.next() {
+    while let Some(item_rslt) = iter.next() {
+        let item = match item_rslt {
+            Err(err) => {
+                mem::forget(guard);
```

Comment:
This `mem::forget` still looks wrong, as per https://github.com/rust-lang/rust/pull/75644#discussion_r619603504

I think because `guard` implements drop you can simply remove this call.

```suggestion
```

Reply:
Oopps... My bad. Thanks!